packages/components/text/src/text.ts

```diff
@@ -0,0 +1,24 @@
+import { buildProps } from '@element-plus/utils'
+import { useSizeProp } from '@element-plus/hooks'
+import type Text from './text.vue'
+import type { ExtractPropTypes } from 'vue'
+
+export const textProps = buildProps({
+  as: {
+    type: String,
+    values: ['span', 'p', 'i', 'u', 'del'],
```

Comment:
## Question

<img width="576" alt="image" src="https://user-images.githubusercontent.com/27342882/221068415-0c6103ed-3371-4066-a898-0b54c9692ad4.png">

There are more tags. Is there a reason for not supporting it?

Reply:
Yes you are right, HTML Formating should support.

After your reminder, I think of more, such as text related HTML tags:
- default
  - span
- [HTML Formatting](https://www.w3schools.com/html/html_formatting.asp)
  - b
  - strong
  - i
  - em
  - mark
  - ~~small~~ (conflict with type prop)
  - ~~abbr~~ (conflict with el-tooltip)
  - del
  - ins
  - cite
  - kbd
  - samp
  - sub
  - sup
- [HTML Semantics](https://www.w3schools.com/html/html5_semantic_elements.asp)
  - p
  - s
  - h1~6
  - time
  - abbr
- [Functional text](https://www.w3schools.com/tags/default.asp) (find text keywords)
  - pre
  - ~~bdi~~ (better support el-config-provider)
  - ~~bdo~~ (better support el-config-provider)

**I'm concerned, too many enums, confusing element-plus user**. We may want to discuss requirements other than technology, `design principle`, `product direction` or `element-plus user needs`.

Reply:
In fact, specifying each tag may be unnecessary. So I think it would be better to define it concisely as below and delegate it to the user.

```
  props: {
    as: {
      type: String,
      default: 'span'
    },
  },
```

What do you think?

Reply:
Good idea. like this?

packages/components/text/src/text.ts
```diff
  props: {
    as: {
      type: String,
-     values: ['span', 'p', 'i', 'u', 'del'],
      default: 'span',
    },
  },
```

docs/en-US/component/text.md
```diff
  | Name | Description          | Type                                          | Default |
- | as   | override the element | ^[enum]`'span' \| 'p' \| 'i' \| 'u' \| 'del'` | span    |
+ | as   | override the element | ^[string]                                     | span    |
```

docs/examples/text/override.vue
```vue
<template>
  <el-space direction="vertical">
    <el-text>span</el-text>
    <el-text as="p">This is a paragraph.</el-text>
    <el-text as="b">Bold</el-text>
    <el-text as="i">Italic</el-text>
    <el-text as="sub">Subscripted</el-text>
    <el-text as="sup">Superscripted</el-text>
    <el-text as="ins">Inserted</el-text>
    <el-text as="del">Deleted</el-text>
    <el-text as="mark">Marked</el-text>
  </el-space>
</template>
```
> Exclude em, strong, small from HTML Fromating tags, em as i, strong as b, recommended to use size prop not small tag.

Reply:
good